### ISSUE TITLE
Suggests to initialize struct with private fields, instead of looking for typo

### ISSUE THREAD

**Issue by nagisa** [rust-lang/rust#39226]
I typo’d `handle` (local variable) with `Handle` (type with private fields I have no access to). Would have expected rustc to not emit such a note and maybe look for typos instead.

```
error[E0423]: expected value, found struct `Handle`
   |
73 |                     handle: Handle,
   |                             ^^^^^^ did you mean `Handle { /* fields */ }`?
```

**Comment by Freyskeyd** [comment]
I agree. I think it's better to look for context before looking for Struct or any other Expression.

I will try to modify check order.

**Comment by Freyskeyd** [comment]
This PR can be closed I think: refer to #39291 

cc @nagisa

**Comment by nagisa** [comment]
Great!
